client_info.go

```diff
@@ -23,7 +24,34 @@ type ClientInfo struct {
 		Version string
 	}
 
-	comment []string
+	Comment []string
+}
+
+// Append returns a new copy of the combined ClientInfo structs
+func (a ClientInfo) Append(b ClientInfo) ClientInfo {
```

Comment:
Q: why not just append with existing products and comments instead of creating new struct all together? any specific rationale for this?

e.g: 
```
a.Products = append(a.Products, b.Products)
a.Comments = append(a.Comments, b.Comments)
return a
```

Any rationale for keeping the original state (a) immutable?